src/covid_model_seiir_pipeline/regression/data.py

```diff
@@ -162,24 +147,23 @@ def load_covariates(self, covariates: Iterable[str], location_ids: List[int]) ->
     ############################
 
     def save_beta_param_file(self, df: pd.DataFrame, draw_id: int) -> None:
-        self.regression_marshall.dump(df, key=MKeys.parameter(draw_id))
+        io.dump(df, self.regression_root.parameters(draw_id=draw_id))
 
     def save_date_file(self, df: pd.DataFrame, draw_id: int) -> None:
-        self.regression_marshall.dump(df, key=MKeys.date(draw_id))
+        io.dump(df, self.regression_root.dates(draw_id=draw_id))
```

Comment:
Ok so, if I'm reading this correctly you rejiggered a whole slew of things but it all works about the same right?

IIRC when I wrote the first version of this keeping a reference to e.g., `self.regression_marshall` was necessary because that object needed to  hold on to an open file handle as most of the backends were stateful.

Is it the case that you're still doing that, or does this do a full write/flush to disk each time?

Reply:
Yes.  Actual usage is pretty close to the same.  The rejiggering is all about unifying the keys and paths abstraction and shifting the burden of state representation to the key-like objects.  

In this situation we are doing a full flush to disk each time.

Reply:
My understanding of your original implementation was that it was flushing to disk on `dump` operations as well.  the state it was managing wasn't the file handles themselves, but just the paths.

Reply:
Upon reading the code you're right - it's always flushed each time.